lib/membrane_http_adaptive_stream/manifest/track.ex

```diff
@@ -26,13 +26,16 @@ defmodule Membrane.HTTPAdaptiveStream.Manifest.Track do
       :segment_duration,
       :partial_segment_duration,
       :header_naming_fun,
-      :segment_naming_fun
+      :segment_naming_fun,
+      :resolution,
+      :maximal_framerate
     ]
     defstruct @enforce_keys ++
                 [
                   target_window_duration: :infinity,
                   persist?: false,
-                  mode: :vod
+                  mode: :vod,
+                  encoding: []
```

Comment:
All of those new fields have default values so either all of them should be enforced or not.